### ISSUE TITLE
Bad parse causes crash

### ISSUE THREAD

**Issue by lihaoyi** [lihaoyi/Ammonite#53]
Similar to https://github.com/lihaoyi/Ammonite/issues/51, but in this case it *should* fail, just more gracefully than it ends up failing

```scala
@ type x = {def t: Int = 1}
java.util.NoSuchElementException: Either.left.value on Right
	scala.util.Either$LeftProjection.get(Either.scala:289)
	ammonite.repl.interp.Preprocessor$$anon$2.apply(Preprocessor.scala:91)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:36)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:32)
	ammonite.repl.Catching.flatMap(Util.scala:73)
	ammonite.repl.interp.Interpreter.processLine(Interpreter.scala:32)
	ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:44)
	ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:43)
	ammonite.repl.frontend.Scoped$$anonfun$flatMap$1.apply(Signaller.scala:43)
	ammonite.repl.frontend.Signaller.apply(Signaller.scala:28)
	ammonite.repl.frontend.Scoped$class.flatMap(Signaller.scala:43)
	ammonite.repl.frontend.Signaller.flatMap(Signaller.scala:10)
	ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:43)
	ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:42)
	ammonite.repl.Res$Success.flatMap(Util.scala:22)
	ammonite.repl.Repl.action(Repl.scala:42)
	ammonite.repl.Repl.loop$1(Repl.scala:53)
	ammonite.repl.Repl.run(Repl.scala:56)
	ammonite.repl.Repl$.main(Repl.scala:80)
	ammonite.repl.Repl.main(Repl.scala)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:483)
	sbt.Run.invokeMain(Run.scala:67)
	sbt.Run.run0(Run.scala:61)
	sbt.Run.sbt$Run$$execute$1(Run.scala:51)
	sbt.Run$$anonfun$run$1.apply$mcV$sp(Run.scala:55)
	sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	sbt.Logger$$anon$4.apply(Logger.scala:85)
	sbt.TrapExit$App.run(TrapExit.scala:248)
	java.lang.Thread.run(Thread.java:745)
Something unexpected went wrong =(
```

**Comment by alexarchambault** [comment]
Getting the same thing with more trivial errors:
```scala
@ bal n = 2
java.util.NoSuchElementException: Either.left.value on Right
        scala.util.Either$LeftProjection.get(Either.scala:289)
        ammonite.repl.interp.Preprocessor$$anon$2.apply(Preprocessor.scala:90)
        ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:37)
        ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:33)
        ammonite.repl.Catching.flatMap(Util.scala:73)
        ammonite.repl.interp.Interpreter.processLine(Interpreter.scala:33)
        ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:45)
        ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:44)
        ammonite.repl.frontend.Scoped$$anonfun$flatMap$1.apply(Signaller.scala:43)
        ammonite.repl.frontend.Signaller.apply(Signaller.scala:28)
        ammonite.repl.frontend.Scoped$class.flatMap(Signaller.scala:43)
        ammonite.repl.frontend.Signaller.flatMap(Signaller.scala:10)
        ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:44)
        ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:43)
        ammonite.repl.Res$Success.flatMap(Util.scala:22)
        ammonite.repl.Repl.action(Repl.scala:43)
        ammonite.repl.Repl.loop$1(Repl.scala:54)
        ammonite.repl.Repl.run(Repl.scala:57)
        ammonite.repl.Repl$.run(Repl.scala:83)
        ammonite.repl.Repl$.main(Repl.scala:63)
        ammonite.repl.Repl.main(Repl.scala)
        sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        java.lang.reflect.Method.invoke(Method.java:497)
        sbt.Run.invokeMain(Run.scala:67)
        sbt.Run.run0(Run.scala:61)
        sbt.Run.sbt$Run$$execute$1(Run.scala:51)
        sbt.Run$$anonfun$run$1.apply$mcV$sp(Run.scala:55)
        sbt.Run$$anonfun$run$1.apply(Run.scala:55)
        sbt.Run$$anonfun$run$1.apply(Run.scala:55)
        sbt.Logger$$anon$4.apply(Logger.scala:85)
        sbt.TrapExit$App.run(TrapExit.scala:248)
        java.lang.Thread.run(Thread.java:745)
Something unexpected went wrong =(
```

**Comment by lihaoyi** [comment]
Fixed by 4826fdbc945933dc375d71d01662ab7daf349b07
